### PR TITLE
Upgrade to Node 16

### PR DESCRIPTION
**Pull request by connor-baer**
## Purpose

[Node 16](https://nodejs.org/en/about/releases/) has entered the long-term support (LTS) phase. 

## Approach and changes

- Upgrade to Node 16 for local development. Consumers of Circuit UI can continue to use older versions of Node.

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: packages/create-sumup-next-app/README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ A
 
 ## 💻 Prerequisites
 
-- You should have **Node.js** installed at a version equal or above **`v10.13.0`**. If you cannot do that, see the Troubleshooting section for ways around this requirement.
+- You should have **Node.js** installed at a version equal or above **`v14.17.0`**. If you cannot do that, see the Troubleshooting section for ways around this requirement.
```
Comment (connor-baer) [inline packages/create-sumup-next-app/README.md:14]: Node 10 is no longer maintained. Foundry v5 will require Node v14.17.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ To start contributing to SumUp Open Source projects, please accept our [Contribu
 
 ### Prerequisites
 
-- [Node.js v14](https://nodejs.org/)
+- [Node.js v16](https://nodejs.org/)
 - [Yarn 1](https://classic.yarnpkg.com/en/docs/install)
```
[the remark this answers is not in the document]
Comment (robinmetral) [reply CONTRIBUTING.md:36]: Well, it'll be unblocked after we're on Node 16 🙂

[the remark this answers is not in the document]
Comment (robinmetral) [reply CONTRIBUTING.md:36]: (npm workspaces are from npm 7 if I recall correctly)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -5,11 +5,14 @@ on: [push]
 jobs:
   ci:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [14, 16]
     steps:
-      - name: Setup Node.js v14
+      - name: Setup Node.js ${{ matrix.node }}
```
Comment (robinmetral) [inline .github/workflows/ci.yml:12]: You read my mind 👀

**Review by robinmetral — APPROVED** [review]
Wonderful 💖

File: .github/workflows/ci.yml (hunk 3 above)
Comment (robinmetral) [reply .github/workflows/ci.yml:12]: Ah I thought you did it before that 😛

**Pull request merged**